app/controllers/entities_controller.rb

```diff
@@ -0,0 +1,36 @@
+class EntitiesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_group
+
+  def index
+    @entities = @group.entities.order(created_at: :desc)
+    @total_amount = @entities.sum(:amount)
+  end
+
+  def new
+    @entity = Entity.new
+  end
+
+  def create
+    @entity = @group.entities.build(entity_params)
+    # @entity.author = current_user
+
+    if @entity.save
+      redirect_to group_entities_path(@group), notice: 'Transaction added successfully!'
+    else
+      render :new
+    end
+  end
```

Comment:
[OPTIONAL]
- [ ] I'm not able to add a new entity to a group via your application. Kindly check it
<video src="https://github.com/EvansSnave/my_budget_app/assets/91142432/d3e28230-85b2-47e4-b970-d8de5cc9995f" />

_**This is a requirement of the project but the first reviewer didn't notice it. so I requested this as an optional but I highly recommend you to make the change**_